src/test/scala/mesosphere/marathon/api/akkahttp/v2/AppsControllerTest.scala

```diff
@@ -1908,5 +1904,105 @@ class AppsControllerTest extends UnitTest with GroupCreation with ScalatestRoute
         header[Headers.`Marathon-Deployment-Id`] should not be 'empty
       }
     }
+
+    "Replace multiple existing applications using Patch" in new Fixture {
+      Given("An app and group")
+      val app1Id = PathId("/app1")
+      val app2Id = PathId("/app2")
+
+      val newApp1Cmd = "bla1"
+      val newApp2Cmd = "bla2"
+      val appDefs = Seq(
+        AppDefinition(id = app1Id, cmd = Some("cmd1")),
+        AppDefinition(id = app2Id, cmd = Some("cmd1"))
+      )
+
+      val rootGroup = createRootGroup(
+        appDefs.map { appDef =>
+        appDef.id -> appDef
+      }.toMap
+      )
+      val plan = DeploymentPlan(rootGroup, rootGroup)
+
+      groupManager.updateRoot(any, any, any, any, any) answers { args =>
+        val fn = args(1).asInstanceOf[RootGroup => RootGroup]
+        val updated = fn(rootGroup)
+        updated.app(app1Id).get.cmd.get shouldEqual newApp1Cmd
+        updated.app(app2Id).get.cmd.get shouldEqual newApp2Cmd
```

Comment:
🤔  hm, I'm not sure I like the assertions here. This diff is super close but I'm wondering if we should rather match the arguments. What do you think? What are we testing here?

Reply:
We're testing that original commands in the new root group was replaced by provided ones. I think without this the test won't make much sense at all :) But we can refactor this once we introduce a 'service' layer.

Reply:
I agree that these assertion should be there. However, I'd like them to be not in a mock. Is there not a way to match the arguments? Cause that's what you do. right? What is the argument anyways?

```scala
val fn = args(1).asInstanceOf[RootGroup => RootGroup]
val updated = fn(rootGroup)
```

seems a little odd. So the first arguments if cast to a function?